src/libstore/remote-store.cc

```diff
@@ -226,6 +226,23 @@ void RemoteStore::initConnection(Connection & conn)
             conn.daemonNixVersion = readString(conn.from);
         }
 
+        if (GET_PROTOCOL_MINOR(conn.daemonVersion) >= 35) {
+            auto temp = readNum<uint8_t>(conn.from);
+            switch (temp) {
+                case 0:
+                    conn.remoteTrustsUs = std::nullopt;
+                    break;
+                case 1 ... 2:
+                    conn.remoteTrustsUs = (TrustedFlag)(temp - 1);
```

Comment:
Let's not try to be confusingly smart here

```suggestion
                case 1:
                    conn.remoteTrustsUs = NotTrusted;
                    break;
                case 2:
                    conn.remoteTrustsUs = Trusted;
```

Reply:
(also, I'd rather move this to a proper `worker_proto::read(..., TrustedFlag)` function like the rest)